Test for RepeatDataset

feihugis opened the pull request:
This PR adds the test for the kernel of `RepeatDataset`.

cc @rachellim @jsimsa

feihugis committed:
Add tests for RepeateDataset kernel

feihugis committed:
Fix the typos in ConcatenateDataset test

jsimsa commented on tensorflow/core/kernels/data/repeat_dataset_op_test.cc:
```diff
@@ -0,0 +1,367 @@
+  NodeDef node_def_;
+};
+
+struct TestParam {
```
s/TestParam/TestCase

jsimsa commented on tensorflow/core/kernels/data/repeat_dataset_op_test.cc:
```diff
@@ -0,0 +1,367 @@
+};
+
+// Test case 1: finite repetition.
+TestParam TestCase1() {
```
s/TestCase1/FiniteRepeatTestCase/ ... and similarly for the other test cases

jsimsa commented on tensorflow/core/kernels/data/repeat_dataset_op_test.cc:
```diff
@@ -0,0 +1,367 @@
+    return Status::OK();
+  }
+
+  Status CreateDatasetFromTestCase(const TestParam &test_case) {
```
no need for `FromTestCase`

jsimsa commented on tensorflow/core/kernels/data/repeat_dataset_op_test.cc:
```diff
@@ -0,0 +1,367 @@
+    return Status::OK();
+  }
+
+  Status CreateIteratorFromTestCase(const TestParam &test_case) {
```
no need for `FromTestCase`

jsimsa commented on tensorflow/core/kernels/data/repeat_dataset_op_test.cc:
```diff
@@ -0,0 +1,367 @@
+    return Status::OK();
+  }
+
+  std::unique_ptr<OpKernel> dataset_kernel_;
```
Please avoid storing this state in member variables. There is no need to do this and it hurts readability.

jsimsa commented on tensorflow/core/kernels/data/repeat_dataset_op_test.cc:
```diff
@@ -0,0 +1,367 @@
+  std::vector<Tensor> out_tensors;
+
+  if (test_case.count < 0) {
+    int fake_infinite_repetition = 100;
```
instead of a cryptic variable name, add a comment here that says that we test only test a finite number of steps of the infinite sequence, and then use a for loop.

jsimsa reviewed (changes requested):
@feihugis thank you for the PR, I did a pass and left some comments

feihugis committed:
Address the comments
- Change the class name and function name to improve readability
- Remove dataset_kernel_ member variable
- Refactor the forever repeat test

feihugis commented:
@jsimsa Thanks for your review! The comments are addressed in [this commit](https://github.com/tensorflow/tensorflow/pull/26415/commits/24ca9eaadf33d5f7b7a1ccaa5d1eb8e2396a02a5). Could you have a look when you have time?

jsimsa commented on tensorflow/core/kernels/data/repeat_dataset_op_test.cc:
```diff
@@ -142,26 +141,26 @@ class RepeatDatasetOpTestHelper : public RepeatDatasetOpTest {
   }
 
-  std::unique_ptr<OpKernel> dataset_kernel_;
   std::unique_ptr<OpKernelContext> dataset_kernel_ctx_;
```
My comment was meant for all of the internal state (`dataset_kernel_`, `dataset_kernel_ctx_`, and `iterator_ctx_`).

While writing the code this way makes it less verbose, it makes it harder to understand what is going on in case someone would like to modify the code.

jsimsa reviewed (changes requested):
@feihugis thank you for the changes, one more comment

feihugis committed:
Improve the code readability

feihugis commented:
@jsimsa [This commit](https://github.com/tensorflow/tensorflow/pull/26415/commits/4ac1f45709d6c251ec3290cc62b67cc4773947e4) removes the internal member variables, and move the code in the helper class to each test to improve the code readability. Could you have a look at the changes? Any suggestions are very welcome.
